File: RepoTransBench/target_projects/Python/C++/tonyo_pyope/tests/test_errors.cpp
#include <gtest/gtest.h>
#include "errors.h"

TEST(Errors, InvalidCiphertextError) {
    EXPECT_THROW(
        throw InvalidCiphertextError("Cipher error");
    , InvalidCiphertextError);
}

TEST(Errors, InvalidRangeLimitsError) {
    EXPECT_THROW(
        throw InvalidRangeLimitsError("Range error");
    , InvalidRangeLimitsError);
}

TEST(Errors, OutOfRangeError) {
    EXPECT_THROW(
        throw OutOfRangeError("Out of range");
    , OutOfRangeError);
}

TEST(Errors, NotEnoughCoinsError) {
    EXPECT_THROW(
        throw NotEnoughCoinsError("No coins left");
    , NotEnoughCoinsError);
}

TEST(Errors, InvalidCoinError) {
    EXPECT_THROW(
        throw InvalidCoinError("Invalid coin");
    , InvalidCoinError);
}

File: RepoTransBench/target_projects/Python/Rust/tonyo_pyope/tests/original/test_errors.rs
use crate::errors::*;

#[test]
fn test_invalid_ciphertext_error() {
    let res = std::panic::catch_unwind(|| {
        panic!("{}", InvalidCiphertextError("Cipher error".to_string()));
    });
    assert!(res.is_err());
}

#[test]
fn test_invalid_range_limits_error() {
    let res = std::panic::catch_unwind(|| {
        panic!("{}", InvalidRangeLimitsError("Range error".to_string()));
    });
    assert!(res.is_err());
}

#[test]
fn test_out_of_range_error() {
    let res = std::panic::catch_unwind(|| {
        panic!("{}", OutOfRangeError("Out of range".to_string()));
    });
    assert!(res.is_err());
}

#[test]
fn test_not_enough_coins_error() {
    let res = std::panic::catch_unwind(|| {
        panic!("{}", NotEnoughCoinsError("No coins left".to_string()));
    });
    assert!(res.is_err());
}

#[test]
fn test_invalid_coin_error() {
    let res = std::panic::catch_unwind(|| {
        panic!("{}", InvalidCoinError("Invalid coin".to_string()));
    });
    assert!(res.is_err());
}